In [1]:
import torch
import torchvision
import torchvision.models as models
from PIL import Image, ImageFile
import numpy as np
import matplotlib.pyplot as plt
import os
import cv2
import gc; gc.enable()
import pandas as pd
from torch.utils.data import Dataset
import tensorflow as tf
from tensorflow import keras
from torchvision import transforms

In [2]:
# filter warnings
import warnings
warnings.simplefilter(action="ignore", category=FutureWarning)
# keras imports
from keras.applications.xception import Xception, preprocess_input
from keras.applications.resnet50 import ResNet50, preprocess_input
from keras.applications.inception_resnet_v2 import InceptionResNetV2, preprocess_input
from keras.applications.mobilenet import MobileNet, preprocess_input
from keras.applications.inception_v3 import InceptionV3, preprocess_input
from keras.preprocessing import image
from keras.models import Model
from keras.models import Sequential
from keras.models import model_from_json
from keras.layers import Input
# other imports
from sklearn.preprocessing import LabelEncoder
import numpy as np
import glob
import cv2
import h5py
import os
import json
import datetime
import time

Using TensorFlow backend.


In [3]:
tf.__version__

'1.13.2'

In [4]:
# Specify image size
IMG_WIDTH = 224
IMG_HEIGHT = 224
CHANNELS = 3

In [5]:
def getRadius(img):
#     print(type(img.shape))
    circles = cv2.HoughCircles(img,3,1,max(img.shape)/2,param1=50,param2=30,minRadius= int(max(img.shape)/6),maxRadius=max(img.shape))
    return circles[0][0][2]
def get_img(id_code):
    img_path = PATH_IMG+id_code+'.png'
    img = cv2.imread(img_path)
#     plt.imshow(img)
    return img

In [6]:
from imgaug import augmenters as ia
import os
import matplotlib.pyplot as plt

PATH_IN =  'data/tr'
PATH_IMG = PATH_IN + "data/train/"

#read in table with image ids and retinopathy severity rating for associated image
df = pd.read_csv("train.csv")
# df = df.head(20).copy()
gc.collect()

# idiot-test with first image
id_code = df['id_code'].values[0]
label = df['diagnosis'].values[0]

In [7]:
#instantiate pre-trained ResNet model
json_file = open('model.json', 'r')
loaded_model_json = json_file.read()
json_file.close()
res_fifty = keras.models.model_from_json(loaded_model_json)
# load weights into new model
res_fifty.load_weights("model.h5")

Instructions for updating:
Colocations handled automatically by placer.


In [8]:
pd.set_option('max_colwidth', -1)

layers = [(layer, layer.name, layer.trainable) for layer in res_fifty.layers]
df_layers = pd.DataFrame(layers, columns=['Layer Type', 'Layer Name', 'Layer Trainable'])
df_layers

Layer Type  \
0    <tensorflow.python.keras.engine.input_layer.InputLayer object at 0x7f24eed1e9b0>             
1    <tensorflow.python.keras.layers.convolutional.ZeroPadding2D object at 0x7f24eed1ee10>        
2    <tensorflow.python.keras.layers.convolutional.Conv2D object at 0x7f24eed34438>               
3    <tensorflow.python.keras.layers.normalization.BatchNormalization object at 0x7f24eed34898>   
4    <tensorflow.python.keras.layers.core.Activation object at 0x7f24eed34c18>                    
..                                                                         ...                    
174  <tensorflow.python.keras.layers.core.Activation object at 0x7f24eec86048>                    
175  <tensorflow.python.keras.layers.pooling.GlobalAveragePooling2D object at 0x7f24eec861d0>     
176  <tensorflow.python.keras.layers.core.Dense object at 0x7f24eec86358>                         
177  <tensorflow.python.keras.layers.core.Dense object at 0x7f24eec86668>                         
178  <tensorflow.python.keras.layers.core.Dense object at 0x7f24eec86978>                         

                   Layer Name  Layer Trainable  
0    input_1                   True             
1    conv1_pad                 False            
2    conv1                     False            
3    bn_conv1                  False            
4    activation                False            
..          ...                  ...            
174  activation_48             True             
175  global_average_pooling2d  True             
176  dense                     True             
177  dense_1                   True             
178  dense_2                   True             

[179 rows x 3 columns]

In [37]:
# from keras import optimizers
datagen = keras.preprocessing.image.ImageDataGenerator(rotation_range = 180,horizontal_flip = True, vertical_flip = True, data_format = 'channels_last')
batch_size = 20
# datagen.fit(ordered_imgs)
 
train_generator = datagen.flow_from_directory(
    'data/train',
    target_size=(224, 224),
    batch_size=batch_size,
    class_mode = 'categorical')


Found 32493 images belonging to 5 classes.


In [38]:
# from keras import backend as K
get_last_pretrained_layer = keras.backend.function([res_fifty.layers[0].input, keras.backend.learning_phase()],
                                  [res_fifty.layers[176].output])


In [39]:
train_features = np.zeros(shape=(32493, 512))
train_labels = np.zeros(shape=(32493,5))

In [41]:
i = 0
for inputs_batch, labels_batch in train_generator:
    features_batch = get_last_pretrained_layer(inputs_batch)[0]
    train_features[i : i + len(inputs_batch)] = features_batch 
    train_labels[i : i +len(inputs_batch)] = labels_batch
    i+= len(inputs_batch)
#     print(i)
    if i+2 > train_features.shape[0]:
        print('beep')
        break
# train_features = np.reshape(train_features, (len(ordered_imgs), 7 * 7 * 512))

ValueError: could not broadcast input array from shape (20,512) into shape (19,512)

In [15]:
del train_generator; gc.collect()

299

In [42]:
i,j = np.where(train_labels==1)

In [43]:
len(j)

31033

In [44]:
sub = train_features[:len(j),:]

In [45]:
sub.shape

(31033, 512)

In [46]:
import xgboost as xgb
from sklearn.svm import SVC as svc
from sklearn.model_selection import train_test_split
# train_features = np.float32(train_features)
# params = {'objective':'multi:softprob'}
features_train, features_test, labels_train, labels_test = train_test_split(sub,j,test_size = 0.2)
clf = svc(kernel = 'poly')
clf.fit(features_train,labels_train)

SVC(C=1.0, cache_size=200, class_weight=None, coef0=0.0,
  decision_function_shape='ovr', degree=3, gamma='auto_deprecated',
  kernel='poly', max_iter=-1, probability=False, random_state=None,
  shrinking=True, tol=0.001, verbose=False)

In [19]:
import pickle
s = pickle.dumps(clf)

In [ ]:
a = pickle.dumps(clf)

In [47]:
preds = clf.predict(features_test)

In [49]:
from sklearn.metrics import cohen_kappa_score
cohen_kappa_score(labels_test,preds, weights = 'quadratic')

0.6759165118919704

polynomial kernel in svm gets higher cohen's kappa than linear kernel and xgboost with "balanced" classes

In [34]:
del train_features, features_train, labels_train; gc.collect()

197

In [21]:
submit = pd.read_csv('../input/aptos2019-blindness-detection/sample_submission.csv')
test_imgs = []
names = {}

In [22]:
from tqdm import tqdm
import matplotlib.pyplot as plt
predicted = []
for i, name in tqdm(enumerate(submit['id_code'])):
    path = os.path.join('../input/aptos2019-blindness-detection/test_images/', name+'.png')
    img_array = cv2.imread(path)
    #crop each image to 80% of fundus diameter
    try:
        r = getRadius(img_array)
        crop_r = round(r*0.8) 
    except:
            crop_r = (img_array.shape[1]/2)*0.8
            
    center_x = int(img_array.shape[1]/2)
    center_y = int(img_array.shape[0]/2)
    
    left_border = int(center_x - crop_r)
    right_border = int(center_x + crop_r)
    upper_border = int(center_y - crop_r)
    lower_border = int(center_y + crop_r)
    
    if left_border < 0:
            difference = 0- left_border
            left_border = 0
            right_border -=difference
            upper_border += difference
            lower_border -= difference
    if upper_border < 0:
            difference = 0- upper_border
            upper_border = 0
            lower_border -= difference
            left_border += difference
            right_border -= difference

    a = cv2.resize(img_array, (224,224), interpolation = cv2.INTER_CUBIC)
    img = cv2.cvtColor(a, cv2.COLOR_BGR2RGB)
    
    test_imgs.append(img)
    score_predict = clf.predict(get_last_pretrained_layer(np.asarray(test_imgs[0:1]))[0])
    predicted.append(score_predict[0])
    test_imgs.pop()

2it [00:00,  6.60it/s]

1
2


4it [00:00,  6.84it/s]

2
2


6it [00:00,  6.96it/s]

4
2


8it [00:01,  5.95it/s]

2
0


10it [00:01,  6.47it/s]

2
0


12it [00:01,  6.08it/s]

1
3


14it [00:02,  6.59it/s]

2
2


15it [00:02,  5.85it/s]

1


17it [00:02,  5.44it/s]

1
2


19it [00:03,  6.21it/s]

2
2


21it [00:03,  6.51it/s]

0
2


23it [00:03,  6.76it/s]

2
1


25it [00:03,  6.96it/s]

2
2


27it [00:04,  7.06it/s]

0
2


29it [00:04,  6.46it/s]

2
1


31it [00:04,  6.77it/s]

1
0


33it [00:05,  6.96it/s]

1
2


35it [00:05,  7.05it/s]

0
2


37it [00:05,  6.96it/s]

1
2


39it [00:05,  7.03it/s]

2
0


40it [00:06,  7.05it/s]

2


42it [00:06,  6.05it/s]

2
2


44it [00:06,  5.56it/s]

4
1


46it [00:07,  5.73it/s]

1
2


48it [00:07,  5.65it/s]

2
3


49it [00:07,  5.89it/s]

2


51it [00:08,  5.51it/s]

1
2


53it [00:08,  6.01it/s]

2
1


55it [00:08,  6.14it/s]

0
1


57it [00:09,  6.56it/s]

0
1


59it [00:09,  6.28it/s]

0
2


61it [00:09,  6.53it/s]

2
1


62it [00:09,  6.49it/s]

2


64it [00:10,  5.80it/s]

2
1


66it [00:10,  6.35it/s]

2
2


68it [00:10,  6.59it/s]

1
2


70it [00:11,  6.79it/s]

1
2


72it [00:11,  5.98it/s]

2
2


73it [00:11,  6.27it/s]

2


75it [00:12,  5.62it/s]

1
2


77it [00:12,  6.39it/s]

1
2


79it [00:12,  6.83it/s]

2
2


81it [00:12,  7.00it/s]

0
0


83it [00:13,  6.92it/s]

2
1


85it [00:13,  6.36it/s]

1
2


87it [00:13,  6.79it/s]

2
2


88it [00:14,  6.68it/s]

1


89it [00:14,  5.50it/s]

1


91it [00:14,  5.63it/s]

1
2


92it [00:14,  5.27it/s]

1


94it [00:15,  5.13it/s]

1
2


96it [00:15,  5.96it/s]

1
2


97it [00:15,  6.04it/s]

1


99it [00:16,  5.64it/s]

1
2


101it [00:16,  6.26it/s]

2
1


103it [00:16,  6.12it/s]

1
0


105it [00:17,  6.33it/s]

2
2


106it [00:17,  6.23it/s]

2


108it [00:17,  5.61it/s]

3
2


110it [00:17,  6.26it/s]

2
1


112it [00:18,  6.07it/s]

2
2


113it [00:18,  5.89it/s]

2


114it [00:18,  4.84it/s]

1


116it [00:19,  5.17it/s]

2
2


118it [00:19,  6.00it/s]

2
2


120it [00:19,  5.88it/s]

1
2


121it [00:19,  5.45it/s]

0


123it [00:20,  5.62it/s]

0
2


124it [00:20,  5.26it/s]

1


126it [00:20,  5.29it/s]

2
2


127it [00:21,  4.71it/s]

2


129it [00:21,  5.15it/s]

1
1


131it [00:21,  5.47it/s]

1
0


133it [00:22,  6.27it/s]

2
2


135it [00:22,  6.49it/s]

2
2


137it [00:22,  6.90it/s]

2
2


139it [00:23,  6.97it/s]

2
0


140it [00:23,  7.03it/s]

1


142it [00:23,  6.03it/s]

1
2


144it [00:23,  6.62it/s]

1
2


146it [00:24,  6.95it/s]

2
2


147it [00:24,  6.27it/s]

0


149it [00:24,  5.46it/s]

1
0


150it [00:24,  5.90it/s]

2


152it [00:25,  5.71it/s]

1
2


154it [00:25,  6.22it/s]

2
0


155it [00:25,  6.47it/s]

1


157it [00:26,  6.14it/s]

1
2


159it [00:26,  6.66it/s]

2
2


161it [00:26,  6.83it/s]

0
1


163it [00:26,  6.89it/s]

2
2


165it [00:27,  6.93it/s]

2
1


167it [00:27,  7.02it/s]

2
0


169it [00:27,  6.35it/s]

4
2


171it [00:28,  6.67it/s]

0
0


173it [00:28,  6.61it/s]

2
4


175it [00:28,  6.83it/s]

0
1


177it [00:28,  6.87it/s]

2
2


179it [00:29,  6.31it/s]

4
0


181it [00:29,  6.64it/s]

0
0


183it [00:29,  6.97it/s]

2
2


184it [00:30,  7.02it/s]

2


186it [00:30,  6.07it/s]

1
1


188it [00:30,  6.19it/s]

0
0


189it [00:30,  6.45it/s]

2


191it [00:31,  5.64it/s]

1
3


193it [00:31,  5.53it/s]

0
4


195it [00:32,  5.48it/s]

4
2


197it [00:32,  6.02it/s]

2
2


199it [00:32,  5.77it/s]

0
2


201it [00:32,  6.39it/s]

2
2


202it [00:33,  6.55it/s]

2


204it [00:33,  5.72it/s]

2
1


206it [00:33,  6.25it/s]

0
0


208it [00:34,  6.00it/s]

2
2


210it [00:34,  6.57it/s]

1
2


211it [00:34,  6.78it/s]

0


213it [00:34,  6.29it/s]

2
2


215it [00:35,  6.69it/s]

2
2


216it [00:35,  6.85it/s]

1


218it [00:35,  6.06it/s]

2
2


220it [00:36,  6.62it/s]

1
2


222it [00:36,  6.90it/s]

2
4


224it [00:36,  6.85it/s]

0
2


226it [00:36,  7.00it/s]

4
2


228it [00:37,  7.14it/s]

2
2


229it [00:37,  7.21it/s]

0


231it [00:37,  6.11it/s]

1
2


232it [00:37,  6.31it/s]

2


233it [00:38,  5.65it/s]

1


234it [00:38,  5.27it/s]

1


236it [00:38,  5.54it/s]

4
1


238it [00:38,  6.17it/s]

2
2


240it [00:39,  5.88it/s]

1
2


242it [00:39,  6.51it/s]

0
2


244it [00:39,  6.73it/s]

1
0


246it [00:40,  6.82it/s]

0
2


248it [00:40,  5.54it/s]

1
2


250it [00:40,  5.63it/s]

0
0


252it [00:41,  5.93it/s]

0
1


253it [00:41,  6.00it/s]

0


255it [00:41,  5.72it/s]

2
2


257it [00:42,  5.71it/s]

1
1


258it [00:42,  5.11it/s]

1


259it [00:42,  4.76it/s]

2


261it [00:43,  5.06it/s]

4
1


263it [00:43,  5.71it/s]

2
0


265it [00:43,  5.91it/s]

0
2


267it [00:43,  6.21it/s]

2
2


268it [00:44,  6.36it/s]

0


270it [00:44,  5.66it/s]

2
1


271it [00:44,  4.96it/s]

1


272it [00:45,  4.71it/s]

1


274it [00:45,  5.02it/s]

1
0


275it [00:45,  5.32it/s]

2


276it [00:45,  5.01it/s]

2


277it [00:46,  4.82it/s]

1


278it [00:46,  4.62it/s]

4


280it [00:46,  5.10it/s]

2
2


282it [00:46,  5.97it/s]

2
2


284it [00:47,  6.40it/s]

2
2


286it [00:47,  6.32it/s]

2
0


288it [00:47,  6.57it/s]

0
0


290it [00:48,  6.87it/s]

4
2


291it [00:48,  6.87it/s]

1


293it [00:48,  6.04it/s]

1
2


295it [00:48,  5.95it/s]

1
2


297it [00:49,  6.34it/s]

0
2


298it [00:49,  6.61it/s]

2


299it [00:49,  5.29it/s]

1


301it [00:50,  5.21it/s]

4
1


303it [00:50,  5.90it/s]

2
2


305it [00:50,  6.49it/s]

2
2


307it [00:50,  6.90it/s]

2
2


308it [00:51,  6.99it/s]

2


310it [00:51,  5.66it/s]

0
1


312it [00:51,  5.98it/s]

2
2


314it [00:52,  6.05it/s]

1
1


316it [00:52,  5.44it/s]

2
2


317it [00:52,  5.15it/s]

1


319it [00:53,  5.29it/s]

4
2


321it [00:53,  6.08it/s]

2
1


323it [00:53,  6.55it/s]

0
1


325it [00:54,  6.82it/s]

2
2


327it [00:54,  6.62it/s]

2
2


329it [00:54,  6.80it/s]

1
2


330it [00:54,  6.84it/s]

2


332it [00:55,  5.87it/s]

2
0


334it [00:55,  5.55it/s]

2
2


336it [00:55,  5.81it/s]

0
2


338it [00:56,  5.66it/s]

1
2


340it [00:56,  6.13it/s]

1
2


342it [00:56,  6.44it/s]

1
2


344it [00:57,  5.76it/s]

0
0


345it [00:57,  5.88it/s]

1


347it [00:57,  5.78it/s]

1
4


349it [00:58,  6.46it/s]

2
0


351it [00:58,  6.20it/s]

2
4


353it [00:58,  6.58it/s]

2
1


355it [00:58,  6.89it/s]

1
2


357it [00:59,  6.92it/s]

1
3


359it [00:59,  6.93it/s]

2
0


361it [00:59,  7.10it/s]

4
1


363it [01:00,  7.12it/s]

2
1


365it [01:00,  6.96it/s]

0
2


367it [01:00,  6.87it/s]

1
1


369it [01:00,  7.06it/s]

1
1


371it [01:01,  7.18it/s]

2
3


373it [01:01,  6.23it/s]

1
1


374it [01:01,  5.19it/s]

2


376it [01:02,  5.37it/s]

1
2


377it [01:02,  4.82it/s]

1


379it [01:02,  5.29it/s]

2
2


381it [01:03,  6.11it/s]

2
2


382it [01:03,  6.42it/s]

2


384it [01:03,  6.05it/s]

1
2


386it [01:03,  6.34it/s]

2
0


388it [01:04,  6.74it/s]

1
2


390it [01:04,  6.99it/s]

2
1


392it [01:04,  6.93it/s]

1
2


394it [01:05,  6.99it/s]

0
2


396it [01:05,  6.68it/s]

2
0


398it [01:05,  6.90it/s]

1
2


400it [01:05,  6.72it/s]

2
0


402it [01:06,  6.98it/s]

2
4


404it [01:06,  7.00it/s]

2
2


406it [01:06,  6.74it/s]

2
2


407it [01:07,  5.67it/s]

1


408it [01:07,  5.31it/s]

2


409it [01:07,  4.81it/s]

2


411it [01:07,  5.26it/s]

0
1


413it [01:08,  5.43it/s]

1
1


415it [01:08,  6.16it/s]

4
1


417it [01:08,  5.60it/s]

1
1


419it [01:09,  6.26it/s]

4
2


421it [01:09,  6.68it/s]

2
2


423it [01:09,  6.55it/s]

2
1


424it [01:09,  6.66it/s]

0


426it [01:10,  6.11it/s]

4
2


428it [01:10,  6.35it/s]

2
2


430it [01:10,  6.52it/s]

1
2


432it [01:11,  6.84it/s]

1
1


433it [01:11,  6.95it/s]

2


434it [01:11,  5.56it/s]

1


436it [01:12,  5.62it/s]

2
2


438it [01:12,  6.38it/s]

2
2


440it [01:12,  6.84it/s]

0
0


441it [01:12,  6.91it/s]

0


443it [01:13,  6.19it/s]

1
0


444it [01:13,  6.48it/s]

1


446it [01:13,  6.14it/s]

1
2


448it [01:13,  6.68it/s]

1
2


450it [01:14,  6.15it/s]

1
0


452it [01:14,  6.59it/s]

1
2


454it [01:14,  6.71it/s]

2
2


456it [01:15,  6.62it/s]

4
2


457it [01:15,  6.73it/s]

4


459it [01:15,  6.17it/s]

2
2


461it [01:15,  6.02it/s]

4
0


463it [01:16,  6.48it/s]

2
2


464it [01:16,  6.66it/s]

1


466it [01:16,  5.79it/s]

1
2


468it [01:17,  6.14it/s]

2
0


470it [01:17,  6.46it/s]

2
2


472it [01:17,  6.41it/s]

1
0


474it [01:17,  6.84it/s]

2
0


475it [01:18,  5.47it/s]

1


477it [01:18,  5.26it/s]

2
2


478it [01:18,  5.66it/s]

1


480it [01:19,  5.17it/s]

2
0


482it [01:19,  5.10it/s]

1
2


484it [01:19,  5.53it/s]

2
2


486it [01:20,  5.28it/s]

1
2


488it [01:20,  5.70it/s]

2
1


490it [01:21,  5.70it/s]

2
2


492it [01:21,  5.66it/s]

2
0


494it [01:21,  5.82it/s]

2
3


496it [01:22,  6.44it/s]

1
2


498it [01:22,  6.27it/s]

2
2


499it [01:22,  6.33it/s]

2


501it [01:22,  6.03it/s]

2
2


502it [01:23,  6.30it/s]

1


504it [01:23,  5.94it/s]

1
2


506it [01:23,  6.47it/s]

2
1


508it [01:23,  6.78it/s]

2
2


510it [01:24,  6.94it/s]

1
2


512it [01:24,  7.03it/s]

2
2


514it [01:24,  7.02it/s]

2
2


516it [01:25,  6.98it/s]

0
2


518it [01:25,  6.96it/s]

2
1


520it [01:25,  6.56it/s]

1
2


522it [01:25,  6.86it/s]

2
4


523it [01:26,  6.89it/s]

2


525it [01:26,  6.25it/s]

1
0


527it [01:26,  6.46it/s]

2
2


529it [01:27,  6.11it/s]

1
2


531it [01:27,  5.64it/s]

2
2


533it [01:27,  6.38it/s]

2
2


535it [01:28,  6.08it/s]

4
2


536it [01:28,  5.55it/s]

1


538it [01:28,  5.73it/s]

1
2


540it [01:29,  6.19it/s]

0
2


542it [01:29,  6.74it/s]

1
2


544it [01:29,  6.83it/s]

2
2


546it [01:29,  7.08it/s]

2
2


548it [01:30,  6.46it/s]

4
2


550it [01:30,  6.87it/s]

2
1


552it [01:30,  6.13it/s]

1
2


554it [01:31,  6.64it/s]

1
2


555it [01:31,  6.84it/s]

2


557it [01:31,  5.79it/s]

1
4


559it [01:32,  6.41it/s]

1
2


561it [01:32,  6.20it/s]

2
1


563it [01:32,  6.61it/s]

2
2


565it [01:32,  6.97it/s]

1
2


567it [01:33,  7.12it/s]

2
2


569it [01:33,  7.19it/s]

2
2


571it [01:33,  6.95it/s]

2
2


573it [01:34,  6.76it/s]

1
2


575it [01:34,  6.98it/s]

2
0


576it [01:34,  7.10it/s]

3


578it [01:34,  6.01it/s]

1
2


580it [01:35,  6.55it/s]

2
0


582it [01:35,  6.57it/s]

2
0


584it [01:35,  6.83it/s]

2
2


586it [01:36,  7.00it/s]

2
2


588it [01:36,  6.96it/s]

0
2


589it [01:36,  7.08it/s]

1


591it [01:36,  5.80it/s]

2
2


593it [01:37,  6.47it/s]

1
0


595it [01:37,  6.87it/s]

2
0


597it [01:37,  6.34it/s]

2
3


599it [01:38,  6.64it/s]

2
2


601it [01:38,  6.83it/s]

0
2


603it [01:38,  6.99it/s]

4
2


605it [01:38,  6.99it/s]

0
2


607it [01:39,  7.03it/s]

0
0


609it [01:39,  7.17it/s]

1
2


610it [01:39,  6.33it/s]

0


612it [01:40,  6.02it/s]

4
2


614it [01:40,  6.58it/s]

2
1


616it [01:40,  6.87it/s]

2
0


618it [01:40,  6.88it/s]

1
2


620it [01:41,  5.89it/s]

1
1


622it [01:41,  5.87it/s]

1
1


624it [01:41,  6.38it/s]

0
2


626it [01:42,  5.74it/s]

1
2


628it [01:42,  6.47it/s]

2
2


630it [01:42,  6.73it/s]

2
1


632it [01:43,  6.80it/s]

2
0


634it [01:43,  6.10it/s]

1
2


635it [01:43,  6.13it/s]

2


637it [01:44,  5.57it/s]

1
2


639it [01:44,  5.76it/s]

2
2


641it [01:44,  6.31it/s]

1
2


643it [01:45,  6.02it/s]

1
2


645it [01:45,  6.25it/s]

2
1


647it [01:45,  5.72it/s]

0
0


649it [01:46,  5.14it/s]

0
2


651it [01:46,  5.46it/s]

4
4


653it [01:46,  6.09it/s]

3
2


655it [01:47,  6.56it/s]

0
2


657it [01:47,  6.44it/s]

1
0


659it [01:47,  6.77it/s]

2
1


661it [01:48,  6.91it/s]

4
2


663it [01:48,  6.86it/s]

2
2


665it [01:48,  6.52it/s]

2
2


667it [01:48,  6.68it/s]

2
2


668it [01:49,  6.69it/s]

2


670it [01:49,  5.99it/s]

2
1


672it [01:49,  6.56it/s]

2
2


674it [01:50,  6.89it/s]

1
2


676it [01:50,  6.39it/s]

2
0


677it [01:50,  6.57it/s]

2


679it [01:50,  6.19it/s]

1
0


681it [01:51,  6.42it/s]

2
0


682it [01:51,  6.63it/s]

2


684it [01:51,  6.09it/s]

2
0


686it [01:52,  6.58it/s]

2
2


688it [01:52,  6.71it/s]

1
2


690it [01:52,  6.62it/s]

2
2


692it [01:52,  6.77it/s]

2
2


694it [01:53,  6.00it/s]

1
2


696it [01:53,  6.48it/s]

0
2


698it [01:53,  6.53it/s]

2
0


700it [01:54,  5.65it/s]

2
1


702it [01:54,  5.66it/s]

1
2


704it [01:54,  6.25it/s]

1
2


706it [01:55,  6.53it/s]

2
1


707it [01:55,  5.38it/s]

0


708it [01:55,  4.74it/s]

2


710it [01:56,  5.11it/s]

2
2


711it [01:56,  4.87it/s]

2


713it [01:56,  5.22it/s]

1
4


715it [01:57,  6.01it/s]

2
2


717it [01:57,  6.38it/s]

0
2


718it [01:57,  6.40it/s]

0


720it [01:57,  6.07it/s]

1
1


722it [01:58,  6.52it/s]

1
2


724it [01:58,  6.75it/s]

0
0


726it [01:58,  6.81it/s]

0
1


728it [01:58,  6.84it/s]

2
1


729it [01:59,  5.89it/s]

1


730it [01:59,  5.30it/s]

2


732it [01:59,  5.33it/s]

4
1


734it [02:00,  6.03it/s]

1
2


735it [02:00,  6.26it/s]

1


737it [02:00,  5.81it/s]

2
2


739it [02:01,  5.74it/s]

1
0


741it [02:01,  6.32it/s]

1
2


743it [02:01,  6.19it/s]

2
2


745it [02:02,  5.66it/s]

2
2


747it [02:02,  6.21it/s]

2
2


749it [02:02,  6.16it/s]

0
2


751it [02:02,  6.67it/s]

2
0


752it [02:03,  6.84it/s]

2


754it [02:03,  6.27it/s]

2
2


756it [02:03,  5.97it/s]

2
2


758it [02:04,  6.53it/s]

1
1


760it [02:04,  6.96it/s]

1
0


762it [02:04,  6.70it/s]

4
2


763it [02:04,  6.79it/s]

2


764it [02:05,  5.96it/s]

1


766it [02:05,  5.87it/s]

2
2


768it [02:05,  6.53it/s]

4
2


769it [02:05,  6.69it/s]

0


771it [02:06,  6.22it/s]

2
1


773it [02:06,  6.74it/s]

2
2


775it [02:06,  6.78it/s]

2
1


777it [02:06,  7.07it/s]

1
2


779it [02:07,  7.18it/s]

2
2


781it [02:07,  7.20it/s]

4
2


783it [02:07,  7.10it/s]

4
1


785it [02:08,  7.12it/s]

2
1


787it [02:08,  7.17it/s]

2
2


788it [02:08,  7.18it/s]

2


790it [02:08,  6.32it/s]

2
1


791it [02:09,  5.24it/s]

1


793it [02:09,  5.48it/s]

1
2


795it [02:09,  5.23it/s]

1
2


797it [02:10,  6.06it/s]

0
2


799it [02:10,  6.60it/s]

2
1


801it [02:10,  6.93it/s]

1
1


803it [02:11,  6.91it/s]

0
2


804it [02:11,  7.03it/s]

2


806it [02:11,  6.39it/s]

2
1


808it [02:11,  6.79it/s]

2
1


810it [02:12,  5.80it/s]

1
4


812it [02:12,  6.47it/s]

0
0


814it [02:12,  6.87it/s]

1
0


816it [02:13,  6.87it/s]

2
2


818it [02:13,  7.01it/s]

2
0


819it [02:13,  7.10it/s]

2


821it [02:13,  6.45it/s]

1
1


823it [02:14,  6.65it/s]

2
0


825it [02:14,  6.15it/s]

4
2


827it [02:14,  5.94it/s]

2
2


829it [02:15,  6.41it/s]

2
2


831it [02:15,  6.82it/s]

2
2


833it [02:15,  6.27it/s]

2
2


835it [02:16,  6.55it/s]

2
2


836it [02:16,  5.45it/s]

0


838it [02:16,  5.67it/s]

2
2


840it [02:17,  5.70it/s]

1
2


842it [02:17,  5.65it/s]

4
2


844it [02:17,  6.11it/s]

2
2


846it [02:18,  6.39it/s]

2
0


848it [02:18,  6.73it/s]

2
1


849it [02:18,  6.88it/s]

1


851it [02:18,  6.36it/s]

0
1


852it [02:18,  6.59it/s]

2


854it [02:19,  6.26it/s]

0
2


855it [02:19,  6.56it/s]

2


857it [02:19,  6.26it/s]

2
2


859it [02:20,  6.61it/s]

0
2


860it [02:20,  5.39it/s]

1


862it [02:20,  5.65it/s]

1
2


864it [02:20,  6.35it/s]

1
1


866it [02:21,  5.80it/s]

1
0


868it [02:21,  6.44it/s]

1
0


869it [02:21,  6.63it/s]

0


871it [02:22,  5.82it/s]

2
2


873it [02:22,  6.43it/s]

2
2


875it [02:22,  6.19it/s]

2
0


876it [02:22,  6.46it/s]

2


878it [02:23,  5.62it/s]

1
2


880it [02:23,  6.26it/s]

0
2


882it [02:23,  6.10it/s]

1
2


884it [02:24,  5.98it/s]

0
2


886it [02:24,  6.54it/s]

2
2


888it [02:24,  6.89it/s]

1
3


890it [02:25,  6.93it/s]

4
2


891it [02:25,  5.91it/s]

1


893it [02:25,  5.31it/s]

1
1


895it [02:26,  6.18it/s]

2
2


896it [02:26,  6.48it/s]

2


898it [02:26,  5.96it/s]

0
2


900it [02:26,  6.60it/s]

2
1


902it [02:27,  6.92it/s]

3
2


904it [02:27,  5.66it/s]

2
2


906it [02:27,  6.35it/s]

2
2


907it [02:28,  6.50it/s]

2


909it [02:28,  5.82it/s]

1
2


911it [02:28,  5.63it/s]

2
1


913it [02:29,  6.36it/s]

2
3


914it [02:29,  6.58it/s]

2


916it [02:29,  5.98it/s]

4
1


918it [02:29,  5.91it/s]

1
0


920it [02:30,  6.51it/s]

2
2


921it [02:30,  6.69it/s]

1


923it [02:30,  5.96it/s]

2
3


925it [02:31,  6.53it/s]

0
2


927it [02:31,  6.88it/s]

2
3


929it [02:31,  6.82it/s]

4
2


931it [02:31,  7.10it/s]

2
2


933it [02:32,  7.18it/s]

2
2


935it [02:32,  7.26it/s]

2
3


937it [02:32,  7.23it/s]

2
2


939it [02:32,  7.21it/s]

2
2


941it [02:33,  7.02it/s]

2
1


943it [02:33,  7.20it/s]

2
1


945it [02:33,  7.15it/s]

2
2


947it [02:34,  7.20it/s]

2
0


948it [02:34,  7.21it/s]

0


950it [02:34,  6.49it/s]

0
2


952it [02:34,  6.73it/s]

2
0


954it [02:35,  6.64it/s]

2
4


955it [02:35,  6.78it/s]

1


956it [02:35,  5.89it/s]

2


958it [02:35,  5.87it/s]

1
2


959it [02:36,  6.20it/s]

2


961it [02:36,  5.73it/s]

2
2


963it [02:36,  6.35it/s]

1
3


965it [02:36,  6.65it/s]

2
4


967it [02:37,  6.25it/s]

1
2


969it [02:37,  6.75it/s]

0
2


971it [02:37,  6.92it/s]

1
0


972it [02:38,  7.03it/s]

2


974it [02:38,  5.93it/s]

2
2


976it [02:38,  6.49it/s]

0
2


977it [02:38,  5.68it/s]

2


978it [02:39,  5.23it/s]

4


979it [02:39,  4.83it/s]

1


980it [02:39,  4.70it/s]

1


982it [02:40,  5.16it/s]

4
1


984it [02:40,  5.99it/s]

2
2


986it [02:40,  6.62it/s]

2
0


988it [02:40,  6.38it/s]

2
0


990it [02:41,  5.87it/s]

1
0


992it [02:41,  6.50it/s]

2
2


994it [02:41,  6.90it/s]

2
2


996it [02:42,  6.86it/s]

2
2


998it [02:42,  6.36it/s]

2
0


1000it [02:42,  6.58it/s]

0
1


1002it [02:43,  5.71it/s]

1
2


1004it [02:43,  6.32it/s]

2
3


1006it [02:43,  6.60it/s]

2
2


1007it [02:43,  6.78it/s]

2


1008it [02:44,  5.89it/s]

2


1010it [02:44,  5.39it/s]

2
1


1012it [02:44,  6.13it/s]

2
1


1014it [02:45,  6.61it/s]

2
2


1015it [02:45,  6.70it/s]

2


1017it [02:45,  5.65it/s]

2
2


1019it [02:45,  6.40it/s]

2
0


1021it [02:46,  6.70it/s]

1
4


1023it [02:46,  6.84it/s]

2
1


1025it [02:46,  7.08it/s]

2
2


1026it [02:46,  7.14it/s]

0


1027it [02:47,  6.14it/s]

0


1029it [02:47,  5.65it/s]

2
2


1031it [02:47,  6.30it/s]

2
1


1032it [02:47,  6.54it/s]

2


1034it [02:48,  6.03it/s]

0
2


1036it [02:48,  6.33it/s]

1
2


1037it [02:48,  6.42it/s]

1


1039it [02:49,  5.91it/s]

2
2


1041it [02:49,  5.34it/s]

4
0


1043it [02:49,  6.11it/s]

2
2


1045it [02:50,  6.47it/s]

0
1


1047it [02:50,  6.57it/s]

0
2


1049it [02:50,  6.58it/s]

0
2


1050it [02:50,  6.73it/s]

2


1051it [02:51,  5.03it/s]

2


1053it [02:51,  5.45it/s]

1
2


1055it [02:51,  6.16it/s]

1
3


1057it [02:52,  6.67it/s]

0
1


1059it [02:52,  6.80it/s]

2
0


1061it [02:52,  6.79it/s]

1
1


1063it [02:53,  6.99it/s]

1
2


1064it [02:53,  7.01it/s]

2


1066it [02:53,  6.43it/s]

0
2


1067it [02:53,  6.64it/s]

2


1069it [02:54,  5.93it/s]

1
0


1071it [02:54,  6.46it/s]

2
2


1073it [02:54,  6.88it/s]

2
2


1075it [02:54,  6.63it/s]

2
2


1076it [02:55,  6.81it/s]

2


1078it [02:55,  5.99it/s]

4
0


1080it [02:55,  6.60it/s]

2
1


1082it [02:55,  6.89it/s]

1
1


1084it [02:56,  6.20it/s]

2
0


1086it [02:56,  6.64it/s]

2
2


1087it [02:56,  6.53it/s]

4


1089it [02:57,  5.79it/s]

0
2


1091it [02:57,  6.21it/s]

2
2


1093it [02:57,  5.71it/s]

4
2


1095it [02:58,  5.82it/s]

0
2


1097it [02:58,  6.25it/s]

2
0


1099it [02:58,  6.35it/s]

1
2


1100it [02:58,  6.46it/s]

2


1102it [02:59,  6.05it/s]

2
1


1104it [02:59,  5.87it/s]

2
0


1106it [02:59,  6.22it/s]

0
2


1108it [03:00,  5.79it/s]

1
2


1110it [03:00,  6.12it/s]

2
2


1112it [03:01,  5.90it/s]

2
2


1114it [03:01,  6.30it/s]

0
2


1115it [03:01,  6.48it/s]

2


1116it [03:01,  4.24it/s]

2


1117it [03:02,  2.95it/s]

0


1119it [03:02,  3.45it/s]

1
0


1121it [03:03,  4.57it/s]

2
0


1123it [03:03,  5.66it/s]

2
2


1125it [03:03,  6.34it/s]

2
3


1127it [03:04,  6.21it/s]

0
2


1129it [03:04,  5.96it/s]

1
0


1130it [03:04,  6.29it/s]

0


1131it [03:04,  5.67it/s]

2


1133it [03:05,  5.51it/s]

2
2


1135it [03:05,  5.63it/s]

2
0


1137it [03:05,  5.76it/s]

2
2


1139it [03:06,  5.98it/s]

0
4


1141it [03:06,  6.51it/s]

2
1


1143it [03:06,  6.61it/s]

3
2


1145it [03:07,  6.80it/s]

2
1


1146it [03:07,  6.30it/s]

0


1148it [03:07,  6.10it/s]

1
2


1150it [03:08,  6.45it/s]

1
1


1152it [03:08,  6.88it/s]

2
2


1154it [03:08,  7.11it/s]

1
1


1156it [03:08,  6.86it/s]

2
2


1158it [03:09,  6.92it/s]

0
2


1160it [03:09,  6.12it/s]

1
2


1162it [03:09,  6.36it/s]

2
4


1164it [03:10,  6.64it/s]

3
2


1166it [03:10,  6.94it/s]

1
3


1168it [03:10,  7.02it/s]

0
2


1169it [03:10,  7.09it/s]

2


1171it [03:11,  6.42it/s]

1
1


1173it [03:11,  6.80it/s]

1
2


1174it [03:11,  6.87it/s]

0


1176it [03:11,  5.90it/s]

2
2


1178it [03:12,  5.64it/s]

1
1


1180it [03:12,  6.37it/s]

0
2


1181it [03:12,  6.63it/s]

2


1183it [03:13,  6.19it/s]

1
1


1184it [03:13,  6.45it/s]

2


1186it [03:13,  6.21it/s]

1
2


1188it [03:13,  6.72it/s]

2
0


1190it [03:14,  6.76it/s]

1
1


1192it [03:14,  6.99it/s]

0
2


1194it [03:14,  6.04it/s]

1
2


1195it [03:15,  6.25it/s]

2


1197it [03:15,  6.06it/s]

2
1


1199it [03:15,  6.31it/s]

4
0


1201it [03:15,  6.77it/s]

2
2


1203it [03:16,  6.93it/s]

1
1


1204it [03:16,  7.06it/s]

2


1206it [03:16,  6.42it/s]

2
2


1208it [03:17,  6.77it/s]

2
2


1210it [03:17,  5.87it/s]

1
2


1212it [03:17,  6.51it/s]

0
0


1214it [03:18,  6.31it/s]

0
2


1216it [03:18,  6.56it/s]

1
2


1218it [03:18,  6.85it/s]

0
2


1220it [03:19,  6.02it/s]

2
4


1222it [03:19,  6.56it/s]

0
1


1224it [03:19,  6.84it/s]

0
2


1226it [03:19,  7.05it/s]

1
3


1228it [03:20,  7.12it/s]

2
1


1230it [03:20,  7.05it/s]

0
0


1231it [03:20,  7.13it/s]

4


1233it [03:20,  6.37it/s]

1
1


1235it [03:21,  6.59it/s]

0
2


1236it [03:21,  6.62it/s]

2


1238it [03:21,  5.62it/s]

2
0


1240it [03:22,  5.85it/s]

1
2


1242it [03:22,  6.42it/s]

0
2


1244it [03:22,  5.92it/s]

2
2


1245it [03:22,  6.04it/s]

2


1247it [03:23,  5.97it/s]

0
2


1249it [03:23,  5.72it/s]

1
1


1251it [03:23,  6.06it/s]

1
1


1253it [03:24,  6.50it/s]

1
1


1255it [03:24,  6.48it/s]

3
2


1257it [03:24,  6.71it/s]

3
1


1258it [03:24,  6.75it/s]

4


1259it [03:25,  5.88it/s]

4


1261it [03:25,  5.09it/s]

1
2


1262it [03:25,  5.57it/s]

2


1264it [03:26,  5.23it/s]

1
4


1265it [03:26,  5.56it/s]

2


1267it [03:26,  5.66it/s]

4
0


1269it [03:27,  6.19it/s]

0
2


1271it [03:27,  5.67it/s]

1
2


1273it [03:27,  6.33it/s]

2
1


1275it [03:27,  6.67it/s]

2
1


1277it [03:28,  5.68it/s]

1
1


1279it [03:28,  6.28it/s]

0
1


1280it [03:28,  6.35it/s]

0


1282it [03:29,  6.14it/s]

0
1


1284it [03:29,  6.19it/s]

1
2


1286it [03:29,  5.94it/s]

2
2


1288it [03:30,  5.98it/s]

2
2


1290it [03:30,  6.53it/s]

2
1


1292it [03:30,  6.43it/s]

1
1


1293it [03:30,  6.57it/s]

0


1295it [03:31,  6.21it/s]

2
2


1297it [03:31,  6.38it/s]

2
2


1299it [03:31,  6.76it/s]

2
2


1300it [03:32,  6.79it/s]

2


1302it [03:32,  6.25it/s]

2
2


1304it [03:32,  6.63it/s]

2
2


1306it [03:33,  6.30it/s]

1
1


1308it [03:33,  6.70it/s]

2
1


1310it [03:33,  6.86it/s]

2
2


1311it [03:33,  6.88it/s]

2


1313it [03:34,  6.03it/s]

1
2


1315it [03:34,  5.91it/s]

1
2


1317it [03:34,  6.39it/s]

0
0


1319it [03:35,  6.80it/s]

1
1


1321it [03:35,  6.90it/s]

0
2


1322it [03:35,  7.00it/s]

2


1323it [03:35,  5.92it/s]

1


1325it [03:36,  5.89it/s]

1
4


1326it [03:36,  5.34it/s]

1


1328it [03:36,  5.60it/s]

2
0


1330it [03:36,  6.29it/s]

1
0


1332it [03:37,  6.17it/s]

1
3


1334it [03:37,  6.67it/s]

1
1


1336it [03:37,  6.85it/s]

0
2


1337it [03:37,  6.95it/s]

2


1338it [03:38,  5.70it/s]

1


1340it [03:38,  5.86it/s]

1
1


1342it [03:38,  6.44it/s]

2
1


1344it [03:39,  6.04it/s]

2
0


1345it [03:39,  6.37it/s]

0


1347it [03:39,  5.66it/s]

4
2


1349it [03:40,  6.27it/s]

2
2


1350it [03:40,  5.64it/s]

1


1352it [03:40,  5.61it/s]

4
4


1354it [03:40,  6.26it/s]

2
0


1355it [03:41,  6.52it/s]

2


1356it [03:41,  5.45it/s]

1


1358it [03:41,  5.60it/s]

2
1


1360it [03:41,  6.22it/s]

1
2


1361it [03:42,  6.46it/s]

2


1363it [03:42,  5.88it/s]

2
2


1365it [03:42,  6.39it/s]

1
3


1367it [03:43,  6.65it/s]

2
2


1369it [03:43,  5.89it/s]

2
2


1371it [03:43,  6.21it/s]

1
0


1372it [03:43,  6.47it/s]

2


1374it [03:44,  5.82it/s]

1
2


1376it [03:44,  6.50it/s]

3
0


1378it [03:44,  6.91it/s]

0
0


1380it [03:45,  6.89it/s]

2
2


1382it [03:45,  6.71it/s]

2
4


1384it [03:45,  6.87it/s]

1
0


1386it [03:45,  6.95it/s]

0
2


1387it [03:46,  6.83it/s]

2


1389it [03:46,  6.02it/s]

2
2


1391it [03:46,  6.58it/s]

1
2


1393it [03:47,  6.43it/s]

2
2


1394it [03:47,  6.66it/s]

2


1395it [03:47,  5.33it/s]

1


1397it [03:47,  5.31it/s]

2
1


1399it [03:48,  6.01it/s]

2
2


1401it [03:48,  5.68it/s]

4
0


1403it [03:48,  5.83it/s]

2
2


1404it [03:49,  6.19it/s]

2


1405it [03:49,  5.60it/s]

1


1407it [03:49,  5.41it/s]

2
2


1408it [03:49,  4.85it/s]

2


1410it [03:50,  5.25it/s]

1
2


1412it [03:50,  6.11it/s]

1
1


1414it [03:50,  5.89it/s]

0
2


1416it [03:51,  6.16it/s]

0
1


1418it [03:51,  6.63it/s]

2
0


1420it [03:51,  6.28it/s]

4
0


1421it [03:52,  5.34it/s]

1


1423it [03:52,  5.61it/s]

0
2


1425it [03:52,  5.73it/s]

3
1


1427it [03:53,  6.21it/s]

4
0


1429it [03:53,  6.62it/s]

1
2


1431it [03:53,  5.92it/s]

2
2


1433it [03:54,  6.50it/s]

1
2


1434it [03:54,  6.68it/s]

2


1436it [03:54,  6.23it/s]

2
0


1438it [03:54,  6.66it/s]

2
0


1439it [03:55,  6.84it/s]

2


1440it [03:55,  5.40it/s]

0


1442it [03:55,  5.56it/s]

4
0


1444it [03:56,  5.37it/s]

2
2


1445it [03:56,  5.11it/s]

1


1446it [03:56,  4.60it/s]

4


1448it [03:56,  5.07it/s]

1
2


1450it [03:57,  5.92it/s]

0
2


1451it [03:57,  6.21it/s]

2


1453it [03:57,  5.85it/s]

2
2


1454it [03:57,  6.16it/s]

3


1456it [03:58,  5.87it/s]

1
1


1458it [03:58,  6.02it/s]

4
2


1460it [03:58,  6.15it/s]

0
1


1462it [03:59,  6.47it/s]

2
0


1463it [03:59,  6.61it/s]

2


1465it [03:59,  5.32it/s]

4
1


1467it [04:00,  5.03it/s]

0
2


1469it [04:00,  5.49it/s]

0
0


1470it [04:00,  5.12it/s]

2


1471it [04:01,  4.80it/s]

1


1472it [04:01,  4.62it/s]

2


1473it [04:01,  4.57it/s]

1


1474it [04:01,  4.61it/s]

2


1475it [04:01,  4.54it/s]

3


1477it [04:02,  5.12it/s]

1
2


1479it [04:02,  5.88it/s]

0
2


1481it [04:02,  6.20it/s]

0
1


1483it [04:03,  6.60it/s]

2
1


1484it [04:03,  5.81it/s]

1


1486it [04:03,  5.51it/s]

2
0


1488it [04:04,  6.24it/s]

2
2


1490it [04:04,  6.71it/s]

2
1


1492it [04:04,  5.72it/s]

1
0


1493it [04:04,  6.05it/s]

2


1495it [04:05,  5.44it/s]

2
0


1497it [04:05,  5.93it/s]

2
2


1499it [04:06,  5.56it/s]

2
2


1501it [04:06,  5.66it/s]

2
2


1502it [04:06,  5.62it/s]

0


1503it [04:06,  5.39it/s]

2


1505it [04:07,  5.08it/s]

2
2


1506it [04:07,  5.12it/s]

1


1507it [04:07,  4.88it/s]

0


1508it [04:07,  4.79it/s]

2


1510it [04:08,  4.90it/s]

1
2


1511it [04:08,  4.91it/s]

1


1512it [04:08,  4.65it/s]

2


1514it [04:09,  4.90it/s]

1
0


1515it [04:09,  5.42it/s]

2


1516it [04:09,  4.89it/s]

1


1518it [04:09,  4.95it/s]

2
2


1520it [04:10,  5.22it/s]

2
1


1522it [04:10,  5.24it/s]

2
2


1524it [04:11,  5.44it/s]

0
2


1526it [04:11,  5.68it/s]

2
2


1528it [04:11,  5.48it/s]

2
0


1530it [04:12,  5.24it/s]

2
1


1531it [04:12,  5.72it/s]

1


1532it [04:12,  5.27it/s]

4


1534it [04:12,  5.59it/s]

2
2


1535it [04:13,  5.21it/s]

1


1537it [04:13,  5.32it/s]

2
1


1539it [04:13,  5.88it/s]

0
2


1541it [04:14,  6.00it/s]

1
2


1542it [04:14,  6.19it/s]

2


1544it [04:14,  5.67it/s]

1
0


1546it [04:15,  5.46it/s]

2
2


1547it [04:15,  5.81it/s]

2


1549it [04:15,  5.84it/s]

1
1


1551it [04:15,  5.53it/s]

2
0


1553it [04:16,  6.22it/s]

2
2


1555it [04:16,  6.20it/s]

0
0


1557it [04:16,  6.73it/s]

1
2


1559it [04:17,  6.61it/s]

1
0


1561it [04:17,  6.89it/s]

2
2


1562it [04:17,  5.90it/s]

2


1564it [04:17,  5.85it/s]

2
2


1566it [04:18,  6.20it/s]

2
2


1568it [04:18,  6.70it/s]

2
2


1570it [04:18,  6.95it/s]

0
1


1572it [04:19,  6.04it/s]

1
0


1574it [04:19,  5.98it/s]

2
0


1576it [04:19,  6.25it/s]

2
1


1578it [04:20,  6.54it/s]

2
0


1580it [04:20,  6.81it/s]

0
2


1582it [04:20,  6.99it/s]

2
4


1584it [04:21,  7.13it/s]

2
4


1585it [04:21,  6.92it/s]

0


1587it [04:21,  6.00it/s]

1
1


1589it [04:21,  6.63it/s]

2
1


1591it [04:22,  6.96it/s]

0
2


1592it [04:22,  7.05it/s]

4


1593it [04:22,  5.70it/s]

1


1595it [04:22,  5.88it/s]

2
1


1597it [04:23,  5.85it/s]

1
3


1599it [04:23,  5.80it/s]

2
2


1601it [04:23,  6.51it/s]

1
2


1603it [04:24,  6.81it/s]

2
2


1605it [04:24,  6.92it/s]

2
2


1607it [04:24,  6.77it/s]

2
0


1608it [04:24,  6.86it/s]

3


1610it [04:25,  6.31it/s]

1
0


1612it [04:25,  6.11it/s]

0
2


1613it [04:25,  6.43it/s]

3


1615it [04:26,  5.80it/s]

1
1


1616it [04:26,  6.02it/s]

1


1618it [04:26,  5.63it/s]

4
3


1620it [04:26,  5.60it/s]

1
2


1621it [04:27,  5.20it/s]

2


1622it [04:27,  5.09it/s]

1


1624it [04:27,  5.08it/s]

1
0


1625it [04:27,  5.57it/s]

2


1627it [04:28,  5.49it/s]

2
2


1629it [04:28,  6.22it/s]

0
2


1631it [04:29,  6.04it/s]

2
0


1633it [04:29,  6.43it/s]

1
2


1634it [04:29,  6.58it/s]

2


1636it [04:29,  6.14it/s]

2
3


1638it [04:30,  6.65it/s]

1
2


1640it [04:30,  6.26it/s]

2
2


1642it [04:30,  6.64it/s]

0
2


1643it [04:30,  6.83it/s]

1


1645it [04:31,  5.89it/s]

1
1


1647it [04:31,  6.41it/s]

1
2


1649it [04:31,  6.68it/s]

2
1


1651it [04:32,  7.00it/s]

2
4


1653it [04:32,  7.17it/s]

1
1


1655it [04:32,  7.12it/s]

1
1


1656it [04:32,  7.04it/s]

0


1658it [04:33,  6.03it/s]

0
4


1660it [04:33,  6.64it/s]

2
2


1662it [04:33,  6.83it/s]

2
1


1664it [04:34,  7.03it/s]

4
2


1666it [04:34,  7.18it/s]

0
0


1668it [04:34,  6.50it/s]

2
3


1670it [04:35,  6.21it/s]

1
1


1672it [04:35,  6.47it/s]

0
2


1674it [04:35,  6.88it/s]

2
1


1676it [04:35,  6.97it/s]

1
0


1678it [04:36,  6.29it/s]

1
0


1679it [04:36,  5.42it/s]

1


1681it [04:36,  5.34it/s]

1
1


1683it [04:37,  5.45it/s]

4
2


1685it [04:37,  6.25it/s]

2
2


1686it [04:37,  5.18it/s]

0


1688it [04:38,  5.49it/s]

1
2


1690it [04:38,  6.31it/s]

0
2


1692it [04:38,  6.73it/s]

2
2


1693it [04:38,  6.87it/s]

1


1695it [04:39,  5.87it/s]

1
2


1697it [04:39,  6.48it/s]

2
2


1699it [04:39,  6.83it/s]

2
2


1701it [04:40,  6.17it/s]

0
0


1703it [04:40,  5.76it/s]

2
0


1705it [04:40,  6.43it/s]

0
2


1707it [04:41,  6.10it/s]

1
2


1708it [04:41,  6.43it/s]

2


1710it [04:41,  6.11it/s]

4
2


1712it [04:41,  6.46it/s]

1
2


1714it [04:42,  5.66it/s]

0
2


1716it [04:42,  6.40it/s]

0
0


1718it [04:42,  6.85it/s]

2
0


1720it [04:43,  6.93it/s]

2
0


1721it [04:43,  6.99it/s]

2


1723it [04:43,  6.08it/s]

1
0


1725it [04:44,  6.43it/s]

2
0


1727it [04:44,  6.84it/s]

0
2


1729it [04:44,  7.02it/s]

2
0


1731it [04:44,  7.04it/s]

2
2


1733it [04:45,  6.32it/s]

2
2


1735it [04:45,  6.67it/s]

2
0


1737it [04:45,  6.34it/s]

1
2


1739it [04:46,  6.57it/s]

1
3


1741it [04:46,  6.94it/s]

2
1


1743it [04:46,  7.12it/s]

2
2


1745it [04:46,  7.15it/s]

2
2


1747it [04:47,  7.09it/s]

2
1


1749it [04:47,  6.97it/s]

0
1


1750it [04:47,  6.04it/s]

2


1752it [04:48,  5.92it/s]

1
2


1754it [04:48,  5.61it/s]

2
1


1755it [04:48,  5.95it/s]

1


1757it [04:49,  5.87it/s]

2
2


1758it [04:49,  6.21it/s]

2


1760it [04:49,  5.69it/s]

1
2


1762it [04:49,  5.86it/s]

0
2


1764it [04:50,  5.54it/s]

2
2


1765it [04:50,  5.14it/s]

1


1767it [04:50,  5.11it/s]

2
0


1769it [04:51,  5.66it/s]

2
2


1771it [04:51,  6.19it/s]

2
1


1773it [04:51,  6.42it/s]

0
0


1775it [04:52,  6.68it/s]

2
0


1777it [04:52,  6.80it/s]

2
2


1779it [04:52,  7.06it/s]

2
2


1781it [04:52,  7.17it/s]

0
2


1782it [04:53,  5.74it/s]

4


1784it [04:53,  5.88it/s]

0
1


1786it [04:53,  6.55it/s]

2
2


1788it [04:54,  6.90it/s]

2
2


1789it [04:54,  6.96it/s]

2


1791it [04:54,  6.23it/s]

2
2


1793it [04:54,  6.74it/s]

2
4


1795it [04:55,  6.18it/s]

2
2


1797it [04:55,  6.50it/s]

1
0


1799it [04:55,  6.75it/s]

2
3


1800it [04:56,  5.47it/s]

2


1802it [04:56,  5.23it/s]

2
0


1804it [04:56,  5.58it/s]

1
0


1805it [04:57,  4.92it/s]

1


1807it [04:57,  5.16it/s]

1
2


1809it [04:57,  5.03it/s]

1
0


1811it [04:58,  5.96it/s]

2
2


1813it [04:58,  5.85it/s]

1
1


1814it [04:58,  6.22it/s]

0


1816it [04:59,  5.84it/s]

1
1


1818it [04:59,  6.49it/s]

2
2


1820it [04:59,  6.65it/s]

0
4


1821it [04:59,  6.86it/s]

2


1823it [05:00,  6.24it/s]

1
2


1824it [05:00,  6.47it/s]

1


1826it [05:00,  6.11it/s]

1
2


1828it [05:00,  6.56it/s]

0
0


1830it [05:01,  6.90it/s]

2
2


1831it [05:01,  6.04it/s]

0


1833it [05:01,  5.63it/s]

0
4


1834it [05:01,  5.99it/s]

2


1836it [05:02,  5.60it/s]

2
1


1838it [05:02,  6.26it/s]

4
4


1840it [05:02,  6.54it/s]

2
1


1841it [05:03,  5.83it/s]

2


1843it [05:03,  5.80it/s]

1
2


1845it [05:03,  6.31it/s]

2
0


1847it [05:04,  6.55it/s]

2
2


1848it [05:04,  6.66it/s]

2


1850it [05:04,  6.05it/s]

2
2


1852it [05:04,  6.49it/s]

2
0


1854it [05:05,  6.85it/s]

1
0


1856it [05:05,  7.03it/s]

1
2


1857it [05:05,  7.12it/s]

2


1859it [05:05,  6.33it/s]

2
1


1861it [05:06,  6.72it/s]

2
0


1863it [05:06,  6.68it/s]

2
1


1865it [05:06,  6.73it/s]

4
1


1866it [05:06,  6.65it/s]

1


1868it [05:07,  5.71it/s]

3
1


1870it [05:07,  5.94it/s]

0
1


1871it [05:07,  5.37it/s]

1


1873it [05:08,  5.53it/s]

2
2


1875it [05:08,  6.04it/s]

2
1


1877it [05:09,  5.59it/s]

3
2


1878it [05:09,  6.03it/s]

2


1880it [05:09,  5.95it/s]

1
2


1882it [05:09,  5.70it/s]

2
0


1884it [05:10,  6.29it/s]

2
1


1886it [05:10,  6.03it/s]

2
4


1888it [05:10,  6.53it/s]

1
2


1890it [05:11,  6.70it/s]

1
1


1891it [05:11,  6.80it/s]

0


1893it [05:11,  6.29it/s]

1
2


1895it [05:11,  6.77it/s]

2
4


1897it [05:12,  6.91it/s]

2
1


1899it [05:12,  7.10it/s]

2
1


1900it [05:12,  7.06it/s]

0


1902it [05:13,  5.95it/s]

0
1


1904it [05:13,  6.20it/s]

2
2


1905it [05:13,  5.65it/s]

1


1906it [05:13,  5.34it/s]

2


1908it [05:14,  5.63it/s]

1
1


1910it [05:14,  6.32it/s]

2
0


1911it [05:14,  6.61it/s]

0


1913it [05:14,  6.15it/s]

2
2


1915it [05:15,  6.48it/s]

2
3


1917it [05:15,  6.84it/s]

2
2


1919it [05:15,  7.05it/s]

1
2


1921it [05:15,  7.23it/s]

2
2


1923it [05:16,  7.13it/s]

2
2


1924it [05:16,  7.13it/s]

0


1926it [05:16,  6.06it/s]

1
1


1928it [05:17,  5.87it/s]

2
0


In [23]:
submit['diagnosis'] = predicted
submit.to_csv('submission.csv', index=False)
submit.head()

id_code  diagnosis
0  0005cfc8afb6  1        
1  003f0afdcd15  2        
2  006efc72b638  2        
3  00836aaacf06  2        
4  009245722fa4  4

occ produces 0.63 accuracy w/ 3662//5 train images, frozen weights

0.68 w/ all train images, last 20 layers unfrozen

0.58 w/ xgboost

0.39 w/ xgboost regressor, Ben's preprocessing algorithm